src/ripple/shamap/tests/SHAMap.test.cpp

```diff
@@ -173,11 +301,14 @@ class SHAMap_test : public beast::unit_test::suite
             keys[7].SetHex ("292891fe4ef6cee585fdc6fda1e09eb4d386363158ec3321b8123e5a772c6ca8");
 
             tests::TestFamily f{beast::Journal{}};
-            SHAMap map{SHAMapType::FREE, f};
+            SHAMap map{SHAMapType::FREE, f, v};
             if (! backed)
                 map.setUnbacked ();
             for (auto const& k : keys)
+            {
                 map.addItem(SHAMapItem{k, IntToVUC(0)}, true, false);
+                map.invariants();
+            }
```

Comment:
It could be made a free function taking a `SHAMap`.  But that wouldn't be my preference.  I really do want to ask the shamap to check itself.  I think it is a good API to have.  I added it to `Ledger` as well.  Once a type is outfitted with an `invariants() const`, it makes debugging a corrupted object so much easier.  You can insert invariant() calls in good and bad places and then bisect the code until you find the line that corrupts.